### PR TITLE
WX-1174 Adjust NIO Copy functionality

### PR DESCRIPTION
**Pull request by JVThomas**
Addresses [WX-1174] (https://broadworkbench.atlassian.net/browse/WX-1174)

PR alters the copy function in the `azure-nio-library` in order to correctly use SAS tokens when making read/copy requests against Azure File Systems (to be used for call caching purposes).

[WX-1174]: https://broadworkbench.atlassian.net/browse/WX-1174?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: runConfigurations/Repo template_ Cromwell server TES.run.xml [hunk 1]
```diff
@@ -16,6 +16,7 @@
     <option name="VM_PARAMETERS" value="-Dconfig.file=target/ci/resources/tes_application.conf" />
     <method v="2">
       <option name="Make" enabled="true" />
+      <option name="RunConfigurationTask" enabled="true" run_configuration_name="renderCiResources" run_configuration_type="SbtRunConfiguration" />
```
Comment (jgainerdewar) [inline runConfigurations/Repo template_ Cromwell server TES.run.xml:19]: 👏 👏

**Review by jgainerdewar — APPROVED** [review]
LGTM!

File: azure-blob-nio/src/main/java/com/azure/storage/blob/nio/AzureFileSystem.java [hunk 2]
```diff
@@ -489,4 +491,11 @@ Long getPutBlobThreshold() {
     Integer getMaxConcurrencyPerRequest() {
         return this.maxConcurrencyPerRequest;
     }
+
+    public String createSASAppendedURL(String url) throws IllegalStateException {
+        if(Objects.isNull(currentActiveSasCredential)) {
+            throw new IllegalStateException("No current active SAS credential present");
+        }
+        return url + "?" + currentActiveSasCredential.getSignature();
```
Comment (aednichols) [inline azure-blob-nio/src/main/java/com/azure/storage/blob/nio/AzureFileSystem.java:499]: Is this supposed to look like
```
https://example.com?credential
```
or
```
https://example.com?key=credential
```

Comment (JVThomas) [reply azure-blob-nio/src/main/java/com/azure/storage/blob/nio/AzureFileSystem.java:499]: Nope, you just simply append the credential to the end of a URI
https://learn.microsoft.com/en-us/azure/storage/common/storage-sas-overview#how-a-shared-access-signature-works

The format's new to me too, I had the same exact reaction when I saw that.

Comment (JVThomas) [reply azure-blob-nio/src/main/java/com/azure/storage/blob/nio/AzureFileSystem.java:499]: Sorry, misread your comment. It's the first option so it'l look like...
`https://example.com?sv=...`

...where the `sv=...` portion is the Azure SAS token

Comment (aednichols) [reply azure-blob-nio/src/main/java/com/azure/storage/blob/nio/AzureFileSystem.java:499]: Makes sense, appreciate the link!

**Review by aednichols — APPROVED** [review]

**Pull request merged**